eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -87,15 +87,15 @@ function update-metadata-table($readmeFolder, $readmeName, $serviceName, $msServ
 {
   $readmePath = Join-Path $readmeFolder -ChildPath $readmeName
   $readmeContent = Get-Content -Path $readmePath -Raw
-  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
+  $null = $readmeContent -match "^---`n*(?<metadata>(((?!---).)*`n)*)---(`n)*(?<content>(.*`n)*)"
```

Comment:
What drove you to change this?

Reply:
We have a readme like:
```
---
title: Azure SDK for Java
author: sima
ms.author: sima
---
...
---
```
Our previous regex is not able to handle this.

Reply:
https://raw.githubusercontent.com/Azure/azure-docs-sdk-java/main/docs-ref-services/latest/hdinsight.md